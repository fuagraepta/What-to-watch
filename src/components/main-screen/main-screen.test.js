import React from 'react';
import renderer from 'react-test-renderer';
import MainScreen from './main-screen.jsx';

describe(`The component is rendered correctly`, () => {
  it(`Main screen correctly renders with transferred mock-movies title`, () => {
    const mockMovies = [
      {
        genre: `Comedies`,
        title: `Johnny English`,
        image: `img/johnny-english.jpg`,
        preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
      },
      {
        genre: `Crime`,
        title: `Snatch`,
        image: `img/snatch.jpg`,
        preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
      },
      {
        genre: `Crime`,
        title: `Pulp Fiction`,
        image: `img/pulp-fiction.jpg`,
        preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
      }
    ];
    const mainScreen = renderer
    .create(<MainScreen
      movies = {mockMovies}
    />)
    .toJSON();

    expect(mainScreen).toMatchSnapshot();
  });
});
