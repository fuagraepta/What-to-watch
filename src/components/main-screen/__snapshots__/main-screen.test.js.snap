// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The component is rendered correctly Main screen correctly renders with transferred mock-movies title 1`] = `
<main>
  <section
    className="movie-card"
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt="The Grand Budapest Hotel"
        src="img/bg-the-grand-budapest-hotel.jpg"
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header movie-card__head"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="user-block"
      >
        <div
          className="user-block__avatar"
        >
          <img
            alt="User avatar"
            height="63"
            src="img/avatar.jpg"
            width="63"
          />
        </div>
      </div>
    </header>
    <div
      className="movie-card__wrap"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster"
        >
          <img
            alt="The Grand Budapest Hotel poster"
            height="327"
            src="img/the-grand-budapest-hotel-poster.jpg"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            The Grand Budapest Hotel
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              Drama
            </span>
            <span
              className="movie-card__year"
            >
              2014
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <button
              className="btn btn--play movie-card__button"
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </button>
            <button
              className="btn btn--list movie-card__button"
              type="button"
            >
              <svg
                height="20"
                viewBox="0 0 19 20"
                width="19"
              >
                <use
                  xlinkHref="#add"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </section>
  <div
    className="page-content"
  >
    <section
      className="catalog"
    >
      <h2
        className="catalog__title visually-hidden"
      >
        Catalog
      </h2>
      <ul
        className="catalog__genres-list"
      >
        <li
          className="catalog__genres-item catalog__genres-item--active"
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            All genres
          </a>
        </li>
        <li
          className="catalog__genres-item"
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Crime
          </a>
        </li>
        <li
          className="catalog__genres-item"
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Comedies
          </a>
        </li>
      </ul>
      <div
        className="catalog__movies-list"
      >
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              style={
                Object {
                  "height": "100%",
                  "objectFit": "fill",
                  "width": "100%",
                }
              }
            />
          </div>
          <h3
            className="small-movie-card__title"
            onClick={[Function]}
          >
            <a
              className="small-movie-card__link"
              href="/films#d90f:fc74:a512:6497:6aab:537:251c:c35a"
            >
              Johnny English
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              style={
                Object {
                  "height": "100%",
                  "objectFit": "fill",
                  "width": "100%",
                }
              }
            />
          </div>
          <h3
            className="small-movie-card__title"
            onClick={[Function]}
          >
            <a
              className="small-movie-card__link"
              href="/films#94b2:9524:2d16:7525:dcb8:9540:7ba0:3afc"
            >
              Snatch
            </a>
          </h3>
        </article>
      </div>
      <div
        className="catalog__more"
      >
        <button
          className="catalog__button "
          onClick={[MockFunction]}
          type="button"
        >
          Show more
        </button>
      </div>
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>
</main>
`;
